CodeJam.Main.Tests/IO/TempDataTests.cs

```diff
@@ -98,6 +112,46 @@ public void TestDirectoryNestedContent()
 			Assert.IsFalse(File.Exists(nestedFile), "File should NOT exist");
 		}
 
+		[Test]
+		public void TestRetryPolicy()
+		{
+			int disposeCallCount = 0;
+			try
+			{
+				Configuration.SetTempDataRetryCallback(
+					a =>
+					{
+						Interlocked.Increment(ref disposeCallCount);
+						a();
+					});
+
+				string dirPath;
+				string nestedFile;
+				using (var dir = TempData.CreateDirectory())
+				{
+					dirPath = dir.Path;
+					nestedFile = Path.Combine(dirPath, "test.tmp");
+
+					using (File.Create(nestedFile))
+					{
+						Assert.Throws<IOException>(() => dir.Dispose());
+						Assert.AreEqual(disposeCallCount, 1);
+					}
+
+					Assert.DoesNotThrow(() => dir.Dispose());
+					Assert.AreEqual(disposeCallCount, 2);
```

Comment:
2 comes first

Reply:
nope, Dispose() call increments disposeCallCount variable

Reply:
I mean

Assert.AreEqual(2, disposeCallCount);

Reply:
nope too. We're using standard order of arguments (`Compare(actual, expected)`) in all tests.

So, there're two options:
1. Rewrite all tests assertions.
2. Do not bother with awkward nUnit design decisions and keep current code as is. 

I strongly prefer second one as it has a very good value to effort ratio :)